x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/AsyncSearchTask.java

```diff
@@ -387,7 +387,7 @@ public void onFinalReduce(List<SearchShard> shards, TotalHits totalHits, Interna
 
         @Override
         public void onResponse(SearchResponse response) {
-            searchResponse.get().updateFinalResponse(response.getSuccessfulShards(), response.getInternalResponse());
+            searchResponse.get().updateFinalResponse(response);
```

Comment:
+1, https://github.com/elastic/elasticsearch/pull/55683 has the same change

Reply:
When I was working on mine I was wondering if I should assert that the final response looks "right" based on our updates. I never did that, but maybe it is good?

Reply:
what do you mean by "right" here?

Reply:
I'm not sure! That was part of why I didn't do it....

I was thinking that it might be useful to know if the response that we got here "agrees" with the results we got from the listener. Maybe that is just checking counts or something.

Reply:
You've already added the assertions below. Ignore me!

Reply:
I see, yea that is the reason why I added the whole response as argument ;)